### PR TITLE
EN 7890 fix cannot simulate tx

### PR DESCRIPTION
**Pull request by iulianpascalau**
- fixed the tx simulation process
- added unit tests for tx simulation rest api endpoint

### CODE REVIEW

**Review by sasurobert — APPROVED** [review]

File: api/transaction/routes_test.go [hunk 1]
```diff
@@ -555,6 +561,216 @@ func TestComputeTransactionGasLimit(t *testing.T) {
 	assert.Equal(t, expectedGasLimit, txCostResp.Data.Cost)
 }
 
+func TestSimulateTransaction_BasRequestShouldErr(t *testing.T) {
```
Comment (bogdan-rosianu) [inline api/transaction/routes_test.go:564]: Bad*

File: facade/interface.go [hunk 2]
```diff
@@ -34,6 +34,7 @@ type NodeHandler interface {
 
 	//ValidateTransaction will validate a transaction
 	ValidateTransaction(tx *transaction.Transaction) error
+	ValidateTransactionForTxSimulation(tx *transaction.Transaction) error
```
Comment (bogdan-rosianu) [inline facade/interface.go:37]: maybe rename to `ValidateSimulatedTransaction`? or `ValidateTransactionForSimulation` ?

File: node/node.go [hunk 3]
```diff
@@ -823,6 +823,31 @@ func (n *Node) ValidateTransaction(tx *transaction.Transaction) error {
 		return err
 	}
 
+	txValidator, intTx, err := n.txCommonValidation(tx)
```
Comment (bogdan-rosianu) [inline node/node.go:826]: could use full words, no abbreviations.

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: api/transaction/routes_test.go (hunk 1 above)
Comment (iulianpascalau) [reply api/transaction/routes_test.go:564]: done

File: facade/interface.go (hunk 2 above)
Comment (iulianpascalau) [reply facade/interface.go:37]: done

File: node/node.go (hunk 3 above)
Comment (iulianpascalau) [reply node/node.go:826]: done

**Review by bogdan-rosianu — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]
👍

**Review by sasurobert — APPROVED** [review]

**Pull request merged**